bluetooth_mesh/gatt.py

```diff
@@ -80,21 +86,51 @@ def read(self, offset=0):
         return self._object.ReadValue(options)
 
     def write(self, value, offset=0):
-        value = GLib.Variant.new_array(
-            None,
-            [GLib.Variant.new_byte(b) for b in value])
-
-        options = {}
-        if offset is not None:
-            options['offset'] = GLib.Variant.new_uint16(offset)
+        if self.write_io:
+            self.write_io.write(value)
+            self.write_io.flush()
+            return
 
-        self._object.WriteValue(value, options)
+        data, fdlist = self._bus.con.call_with_unix_fd_list_sync(
+                                                self._object._bus_name, 
+                                                self._object._path, 
+                                                self._object.AcquireWrite._iface_name, 
+                                                'AcquireWrite',
+                                                GLib.Variant('(a{sv})', [{}]),
+                                                GLib.VariantType.new("(hq)"), 0, -1, None, None
+                                                )
+        fd = fdlist.steal_fds()[data[0]]
+        ch = GLib.IOChannel(filedes=fd)
+        ch.set_encoding(None)
+        ch.write(value)
+        ch.flush()
+        self.write_io = ch
 
     def notify(self, enabled=True):
         if enabled:
-            self._object.StartNotify()
+            notify_data, fdlist = self._bus.con.call_with_unix_fd_list_sync(
+                                                    self._object._bus_name, 
+                                                    self._object._path, 
+                                                    self._object.AcquireNotify._iface_name, 
+                                                    'AcquireNotify',
+                                                    GLib.Variant('(a{sv})', [{}]),
+                                                    GLib.VariantType.new("(hq)"), 0, -1, None, None
+                                                    )
+            fd = fdlist.steal_fds()[notify_data[0]]
+            
+            ch = GLib.IOChannel(filedes=fd)
+            ch.set_encoding(None)
+            ch.set_flags(ch.get_flags() | GLib.IOFlags.NONBLOCK)
+            self.read_io = ch
```

Comment:
Same as `write_io`, this could be a property. Rename to `read_channel` maybe?

Reply:
same as above